spaceship.zsh

```diff
@@ -370,6 +390,71 @@ spaceship_git() {
     "$SPACESHIP_GIT_SUFFIX"
 }
 
+spaceship_hg_branch() {
+  [[ $SPACESHIP_HG_BRANCH_SHOW == false ]] && return
+
+  _is_hg || return
+
+  local hg_branch="$(cat $PWD/.hg/branch)"
+
+  _prompt_section \
+    "$SPACESHIP_HG_BRANCH_COLOR" \
+    "$SPACESHIP_HG_BRANCH_PREFIX"$hg_branch"$SPACESHIP_HG_BRANCH_SUFFIX"
+}
+
+spaceship_hg_status() {
+  [[ $SPACESHIP_HG_STATUS_SHOW == false ]] && return
+
+  _is_hg || return
+
+  local INDEX=$(command hg status 2>/dev/null) hg_status=""
+
+  # Indicators are suffixed instead of prefixed to each other to
+  # provide uniform view across git and mercurial indicators
+  if $(echo "$INDEX" | command grep -E '^\? ' &> /dev/null); then
+    hg_status="$SPACESHIP_HG_STATUS_UNTRACKED$hg_status"
+  fi
+  if $(echo "$INDEX" | command grep -E '^A ' &> /dev/null); then
+    hg_status="$SPACESHIP_HG_STATUS_ADDED$hg_status"
+  fi
+  if $(echo "$INDEX" | command grep -E '^M ' &> /dev/null); then
+    hg_status="$SPACESHIP_HG_STATUS_MODIFIED$hg_status"
+  fi
+  if $(echo "$INDEX" | command grep -E '^(R|!)' &> /dev/null); then
+    hg_status="$SPACESHIP_HG_STATUS_DELETED$hg_status"
+  fi
+
+  # outgoing and incoming compare with remote branches, everytime.
+  # Works only if connected to internet
+  # Increases prompt redraw time considerably
+  if $(hg outgoing 2>/dev/null | grep -E "^changeset" &> /dev/null); then
+    hg_status="$SPACESHIP_HG_STATUS_AHEAD$hg_status"
+  fi
+  if $(hg incoming 2>/dev/null | grep -E "^changeset" &> /dev/null); then
+    hg_status="$SPACESHIP_HG_STATUS_BEHIND$hg_status"
```

Comment:
@denysdovhan These two indicators increase prompt redraw time due to remote comparison. Is there any better methods ?

If not, what about providing option to disable them like `SPACESHIP_HG_STATUS_AHEAD_SHOW` and `SPACESHIP_HG_STATUS_BEHIND_SHOW` ?